client/app/lib/flux/teams/actions.coffee

```diff
@@ -315,21 +316,26 @@ deleteApiToken = (apiTokenId) ->
   reactor.dispatch actions.DELETE_API_TOKEN_SUCCESS, { apiTokenId }
 
 
-addApiToken = (apiToken) ->
+addApiToken = ->
 
   { reactor } = kd.singletons
-  reactor.dispatch actions.ADD_API_TOKEN_SUCCESS, { apiToken }
+  remote.api.JApiToken.create (err, apiToken) ->
+
+    return showError err  if err
+
+    reactor.dispatch actions.ADD_API_TOKEN_SUCCESS, { apiToken }
 
 
 disableApiTokens = (state) ->
 
   { groupsController, reactor } = kd.singletons
   team = groupsController.getCurrentGroup()
-  new Promise (resolve, reject) ->
 
-    team.modify { isApiEnabled : state }, (err) ->
-      return reject { err }  if err
-      reactor.dispatch actions.SET_API_ACCESS_STATE, { state }
+  team.modify { isApiEnabled : state }, (err) ->
+
+    return showError err  if err
```

Comment:
I think there is no need to use Promise here, because we are not returning anything to view component, if there is an error, showError will show it 🤔